cuda_core/cuda/core/experimental/_launcher.py

```diff
@@ -67,24 +90,34 @@ def launch(kernel, config, *kernel_args):
     if not isinstance(kernel, Kernel):
         raise ValueError
     config = check_or_create_options(LaunchConfig, config, "launch config")
+    if config.stream is None:
```

Comment:
This is error-out by design (we want users to be explicit about which stream to use). That comment (line 51) means the check is done in `launch()` (i.e. here) instead of the `LaunchConfig` constructor (including `__post_init__`) because the latter can still have members changed after creation.